src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -48,6 +48,42 @@ protected function tearDown()
         }
     }
 
+    /**
+     * @group legacy
+     * @expectedDeprecation The provided cwd does not exist. Command is currently ran against getcwd(). This behavior is deprecated since version 3.4 and will be removed in 4.0.
+     */
+    public function testInvalidCwd()
+    {
+        if ('\\' === DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('Windows handles this automatically.');
+        }
+
+        // Check that it works fine if the CWD exists
+        $cmd = new Process('echo test', __DIR__);
+        $cmd->run();
+
+        $cmd = new Process('echo test', __DIR__.'/notfound/');
+        $cmd->run();
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Process\Exception\RuntimeException
+     * @expectedExceptionMessage The provided cwd does not exist.
+     */
+    public function testInvalidCwdOnWindows()
+    {
+        if ('\\' !== DIRECTORY_SEPARATOR) {
+            $this->markTestSkipped('Unix handles this automatically.');
+        }
+
+        // Check that it works fine if the CWD exists
```

Comment:
This does not test anything. To test that it works, it must be a separate test.

If the exception happens in the first process rather than the second one, the `@expectedException` assertion will be fine with it.

Reply:
Handled this by forcing a fail if any exception occurs in that first section.